addons/calendar/tests/test_mail_activity_mixin.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+import pytz
+
+from odoo import Command
+from odoo import tests
+from odoo.addons.mail.tests.common import MailCommon
+
+
+@tests.tagged('mail_activity_mixin')
+class TestMailActivityMixin(MailCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestMailActivityMixin, cls).setUpClass()
+        # using res.partner as the model inheriting from mail.activity.mixin
+        cls.test_record = cls.env['res.partner'].with_context(cls._test_context).create({'name': 'Test'})
+        cls.activity_type_1 = cls.env['mail.activity.type'].create({
+            'name': 'Calendar Activity Test Default',
+            'summary': 'default activity',
+            'res_model_id': cls.env['ir.model']._get('base.res_partner').id,
+        })
+        cls.env['ir.model.data'].create({
+            'name': cls.activity_type_1.name.lower().replace(' ', '_'),
+            'module': 'calendar',
+            'model': cls.activity_type_1._name,
+            'res_id': cls.activity_type_1.id,
+        })
+        # reset ctx
+        cls._reset_mail_context(cls.test_record)
+
+    def test_activity_calendar_event_id(self):
+        """ test the computed field "activity_calendar_event_id" which is the event of the
+        next activity. It must evaluate to False if the next activity is not related to an event
+        """
+        def create_event(name, event_date):
+            event_date_str = event_date.strftime('%Y-%m-%d')
+            return self.env['calendar.event'].create({
+                'name': name,
+                'start': event_date_str + ' 12:00:00',
```

Comment:
Can't you use datetime.combine + time ? With something like (did not test, may have to fix / dive into lib :D)
```
'start': datetime.datetime.combine(event_date, datetime.time(12, 0, 0))
```